### DISCUSSION TITLE
How do I implement RunInContainer

### DISCUSSION THREAD

**Discussion by yueyongyue** [virtual-kubelet/virtual-kubelet#1057] [Q&A]
---
Provider  exec API return WebSocket， How do I implement RunInContainer？

### Environment summary

Provider (e.g. ACI, AWS Fargate)

Version (e.g. 0.1, 0.2-beta)

K8s Master Info (e.g. AKS, ACS, Bare Metal, EKS)

Install Method (e.g. Helm Chart, )

**Comment by cpuguy83** [comment]
`RunInContainer` does not deal with any protocol level details, just stdio streams:

https://github.com/virtual-kubelet/virtual-kubelet/blob/704b01eac6bdf0472b9c93173e4cb64bd6e53e94/node/api/exec.go#L37-L43

The api handler in that same file deals with the websocket things.
